Duplicate definitions for Write1

<!-- Thanks for filing a bindgen issue! We appreciate it :-) -->

### Input C/C++ Header
I don't quite understand the header I'm using so I'm not able to make a simple example but I'm using RakNet https://github.com/facebookarchive/RakNet.
The header with these functions is BitStream.h

### Bindgen Invocation
<!-- Place either the `bindgen::Builder` or the command line flags used here. -->

```Rust
    let bindings = bindgen::Builder::default()
        .header("wrapper.hpp")
        .enable_cxx_namespaces()
        .layout_tests(false)

        .allowlist_type("Rak.*")
        .allowlist_type("DefaultMessageIDTypes")

        .generate()
        .expect("Unable to generate bindings");
```

### Actual Results

```
error[E0201]: duplicate definitions with name `Write1`:
    --> /home/bloup/Code/raknet/target/debug/build/raknet-d5cd88364ce5f684/out/bindings.rs:3003:13
     |
2811 | /             pub unsafe fn Write1(
2812 | |                 &mut self,
2813 | |                 bitStream: *mut root::RakNet::BitStream,
2814 | |                 numberOfBits: root::RakNet::BitSize_t,
2815 | |             ) {
2816 | |                 BitStream_Write1(self, bitStream, numberOfBits)
2817 | |             }
     | |_____________- previous definition of `Write1` here
...
3003 | /             pub unsafe fn Write1(&mut self) {
3004 | |                 BitStream_Write11(self)
3005 | |             }
     | |_____________^ duplicate definition
```
### Expected Results

<!--
Replace this with a description of what you expected instead of the actual
results. The more precise, the better! For example, if a struct in the generated
bindings is missing a field that exists in the C/C++ struct, note that here.
-->
I would expect the second Write1 to be Write11 and not cause issues.

--- a/bindgen-tests/tests/expectations/tests/duplicated-definition-count.rs
+++ b/bindgen-tests/tests/expectations/tests/duplicated-definition-count.rs
@@ -1,0 +1,67 @@
+#![allow(
+    dead_code,
+    non_snake_case,
+    non_camel_case_types,
+    non_upper_case_globals
+)]
+
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct BitStream {
+    pub _address: u8,
+}
+#[test]
+fn bindgen_test_layout_BitStream() {
+    assert_eq!(
+        ::std::mem::size_of::<BitStream>(),
+        1usize,
+        concat!("Size of: ", stringify!(BitStream))
+    );
+    assert_eq!(
+        ::std::mem::align_of::<BitStream>(),
+        1usize,
+        concat!("Alignment of ", stringify!(BitStream))
+    );
+}
+extern "C" {
+    #[link_name = "\u{1}_ZN9BitStream5WriteEPKcj"]
+    pub fn BitStream_Write(
+        this: *mut BitStream,
+        inputByteArray: *const ::std::os::raw::c_char,
+        numberOfBytes: ::std::os::raw::c_uint,
+    );
+}
+extern "C" {
+    #[link_name = "\u{1}_ZN9BitStream5WriteEPS_j"]
+    pub fn BitStream_Write1(
+        this: *mut BitStream,
+        bitStream: *mut BitStream,
+        numberOfBits: ::std::os::raw::c_uint,
+    );
+}
+extern "C" {
+    #[link_name = "\u{1}_ZN9BitStream6Write1Ev"]
+    pub fn BitStream_Write11(this: *mut BitStream);
+}
+impl BitStream {
+    #[inline]
+    pub unsafe fn Write(
+        &mut self,
+        inputByteArray: *const ::std::os::raw::c_char,
+        numberOfBytes: ::std::os::raw::c_uint,
+    ) {
+        unsafe { BitStream_Write(self, inputByteArray, numberOfBytes) }
+    }
+    #[inline]
+    pub unsafe fn Write1(
+        &mut self,
+        bitStream: *mut BitStream,
+        numberOfBits: ::std::os::raw::c_uint,
+    ) {
+        unsafe { BitStream_Write1(self, bitStream, numberOfBits) }
+    }
+    #[inline]
+    pub unsafe fn Write11(&mut self) {
+        unsafe { BitStream_Write11(self) }
+    }
+}

--- a/bindgen-tests/tests/headers/duplicated-definition-count.hpp
+++ b/bindgen-tests/tests/headers/duplicated-definition-count.hpp
@@ -1,0 +1,6 @@
+class BitStream {
+ public:
+  void Write(const char *inputByteArray, unsigned int numberOfBytes);
+  void Write(BitStream *bitStream, unsigned numberOfBits);
+  void Write1();
+};

--- a/bindgen/codegen/mod.rs
+++ b/bindgen/codegen/mod.rs
@@ -2421,7 +2421,7 @@ trait MethodCodegen {
         &self,
         ctx: &BindgenContext,
         methods: &mut Vec<proc_macro2::TokenStream>,
-        method_names: &mut HashMap<String, usize>,
+        method_names: &mut HashSet<String>,
         result: &mut CodegenResult<'a>,
         parent: &CompInfo,
     );
@@ -2432,7 +2432,7 @@ impl MethodCodegen for Method {
         &self,
         ctx: &BindgenContext,
         methods: &mut Vec<proc_macro2::TokenStream>,
-        method_names: &mut HashMap<String, usize>,
+        method_names: &mut HashSet<String>,
         result: &mut CodegenResult<'a>,
         _parent: &CompInfo,
     ) {
@@ -2499,15 +2499,21 @@ impl MethodCodegen for Method {
             return;
         }
 
-        let count = {
-            let count = method_names.entry(name.clone()).or_insert(0);
-            *count += 1;
-            *count - 1
-        };
+        if method_names.contains(&name) {
+            let mut count = 1;
+            let mut new_name;
 
-        if count != 0 {
-            name.push_str(&count.to_string());
+            while {
+                new_name = format!("{}{}", name, count);
+                method_names.contains(&new_name)
+            } {
+                count += 1;
+            }
+
+            name = new_name;
         }
+
+        method_names.insert(name.clone());
 
         let mut function_name = function_item.canonical_name(ctx);
         if times_seen > 0 {

